pkg/storage/errors.go

```diff
@@ -68,15 +70,24 @@ func NewUnreachableError(key string, rv int64) *StorageError {
 	}
 }
 
+func NewInvalidObjError(key, msg string) *StorageError {
+	return &StorageError{
+		Code:               ErrCodeInvalidObj,
+		Key:                key,
+		AdditionalErrorMsg: msg,
+	}
+}
+
 type StorageError struct {
-	Code            int
-	Key             string
-	ResourceVersion int64
+	Code               int
+	Key                string
+	ResourceVersion    int64
+	AdditionalErrorMsg string
```

Comment:
See discussion [here](https://github.com/kubernetes/kubernetes/pull/23387#discussion_r57696139).

The issue is that we are uncertain about what to put in for precondition and other minor cases. We might want to make this more generic and extensible at the moment.